vars/neoDeploy.groovy

```diff
@@ -19,33 +19,104 @@ def call(parameters = [:]) {
             error "Archive cannot be found with parameter archivePath: '${archivePath}'."
         }
 
+        def deployMode = utils.getMandatoryParameter(parameters, 'deployMode', 'MTA')
+
+        if (deployMode != 'MTA' && deployMode != 'WAR_PARAMS' && deployMode != 'WAR_PROPERTIESFILE') {
+            throw new IllegalArgumentException("[neoDeploy] Invalid deployMode = '${deployMode}'. Valid 'deployMode' values are: 'MTA', 'WAR_PARAMS' and 'WAR_PROPERTIESFILE'")
+        }
+
+        def propertiesFile
+        def warAction
+        if (deployMode == 'WAR_PROPERTIESFILE' || deployMode == 'WAR_PARAMS') {
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+            if (warAction != 'warAction' && warAction != 'deploy') {
+                throw new IllegalArgumentException("[neoDeploy] Invalid warAction = '${warAction}'. Valid 'warAction' values are: 'deploy' and 'rolling-update'.")
+            }
+        }
+        if (deployMode == 'WAR_PROPERTIESFILE') {
+            propertiesFile = new File(utils.getMandatoryParameter(parameters, 'propertiesFile', null))
+            if (!propertiesFile.isAbsolute()) {
+                propertiesFile = new File(pwd(), propertiesFile.getPath())
+            }
+            if (!propertiesFile.exists()){
+                error "Properties file cannot be found with parameter propertiesFile: '${propertiesFile}'."
+            }
+        }
+
+        def applicationName
+        def runtime
+        def runtimeVersion
+        def vmSize
+        if (deployMode == 'WAR_PARAMS') {
```

Comment:
Could you please convert the parameter values to lower case. Otherwise it would be different to other steps.